src/dbus_api/api/shared.rs

```diff
@@ -31,18 +31,24 @@ use crate::{
 /// the interface.
 pub fn create_pool_shared(
     m: &MethodInfo<MTSync<TData>, TData>,
-    has_key_desc: bool,
+    has_additional_params: Option<bool>,
```

Comment:
I'm happy to define a tri-state enum for this if that seems better to you, but I really hesitate to allow integers when this introduces additional error checking into the callee to ensure that a valid integer has been passed in. I would much prefer a finite set of options at the type level instead of having to do validation on the value of the input.

Reply:
Ok! Do please go with the enum.